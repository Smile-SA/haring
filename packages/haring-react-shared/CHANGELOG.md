# @smile/haring-react-shared

## 0.15.0

### Minor Changes

- [#194](https://github.com/Smile-SA/haring/pull/194) [`97a610a`](https://github.com/Smile-SA/haring/commit/97a610aa02b86423b723ef8ae5d45e8ecf346f92) Thanks [@vapersmile](https://github.com/vapersmile)! - Upgrade mantine version to 7.11.x

## 0.14.0

## 0.13.0

### Minor Changes

- [#178](https://github.com/Smile-SA/haring/pull/178) [`ce8f39a2c52590b2c91950450a84222434913743`](https://github.com/Smile-SA/haring/commit/ce8f39a2c52590b2c91950450a84222434913743) Thanks [@tonai](https://github.com/tonai)! - Rename packages

## 0.12.0

## 0.11.0

## 0.10.0

### Minor Changes

- [#149](https://github.com/Smile-SA/haring/pull/149) [`2706a09`](https://github.com/Smile-SA/haring/commit/2706a097fd86b072cf21d12fe2a97427883671cf) Thanks [@vapersmile](https://github.com/vapersmile)! - Add Calendar style on globlaStyle and add a preview

- [#150](https://github.com/Smile-SA/haring/pull/150) [`34fabfb`](https://github.com/Smile-SA/haring/commit/34fabfbd1a2c048c39adc567548a3ee7e85074ee) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Fixed missing global styles for focus-visible color

## 0.9.0

### Minor Changes

- [`d430e2c`](https://github.com/Smile-SA/haring/commit/d430e2cbcd26871c4db0c53f92a4e4baa6af839c) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Migrated from Mantine V6 to V7

## 0.8.1

## 0.8.0

### Minor Changes

- [#108](https://github.com/Smile-SA/haring/pull/108) [`320634d`](https://github.com/Smile-SA/haring/commit/320634d7ec2048a328cfda1dbcff5d1b70a32563) Thanks [@vapersmile](https://github.com/vapersmile)! - Add custom default style for carousel mantine composent and add 2-Mantine/carousel page

## 0.7.0

### Minor Changes

- [#87](https://github.com/Smile-SA/haring/pull/87) [`dbc06ca`](https://github.com/Smile-SA/haring/commit/dbc06ca55961b69663ab7fdc02609c6525ae361d) Thanks [@tonai](https://github.com/tonai)! - Add theme provider and update barrel file for better next.js compatibility

- [#95](https://github.com/Smile-SA/haring/pull/95) [`5696400`](https://github.com/Smile-SA/haring/commit/5696400e6f703da52db5f7199b50f8251fa76726) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Added `FilterList` component, renamed `FiltersBar` in `SidebarFilters`
  and `FiltersCheckboxList` in `SearchableCheckboxList`,
  refactored `SearchableCheckboxList`, added common `IFilter` type in shared
  package, added topContent prop to `SwitchableView`, fixed `BrowsingPage` with
  name changes.

## 0.6.0

### Minor Changes

- [#74](https://github.com/Smile-SA/haring/pull/74) [`0ad5377`](https://github.com/Smile-SA/haring/commit/0ad5377535e7c19941da301e8a2ee7298ab70f91) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Refactored actions in `ThumbnailGrid` and `TableGridView` components, moved `typeGuard` helper to shared package, added `typeGuardInterface` function, added tests

## 0.5.0

### Minor Changes

- [#72](https://github.com/Smile-SA/haring/pull/72) [`c8d3d94`](https://github.com/Smile-SA/haring/commit/c8d3d946cb0ded4ed26d4b8be249eff45d3d56f6) Thanks [@tonai](https://github.com/tonai)! - Add dynamic componentProps property in actions

- [#65](https://github.com/Smile-SA/haring/pull/65) [`bd9cfe4`](https://github.com/Smile-SA/haring/commit/bd9cfe42d0b22f5f7f5e7b0de30fdfb22ad3e1c8) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Reworked action/confirm types and moved then in `react-front-kit-shared` then refactored types in `Table`, `Thumbnail` and `ConfirmModal` , Added `ThumbnailGrid` and `TableGridView` components,

- [#68](https://github.com/Smile-SA/haring/pull/68) [`c8b5a39`](https://github.com/Smile-SA/haring/commit/c8b5a3978c8ce7133da0b498d9f0b326f07eb737) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Added `storybook-pages` package containing example pages, renamed `TestPage`
  into `BrowsingPage` and completed it with integration of `TableGridView`, fixed
  various component styles and props related to page.

### Patch Changes

- [#72](https://github.com/Smile-SA/haring/pull/72) [`c8d3d94`](https://github.com/Smile-SA/haring/commit/c8d3d946cb0ded4ed26d4b8be249eff45d3d56f6) Thanks [@tonai](https://github.com/tonai)! - Add server barrel files

## 0.4.0

### Minor Changes

- [#53](https://github.com/Smile-SA/haring/pull/53) [`01ee3ea`](https://github.com/Smile-SA/haring/commit/01ee3ea74827cb6e3d4041b50fbab959750135ff) Thanks [@tonai](https://github.com/tonai)! - Use `Paper` for `InfoCard`, update test page and refactor `Pagination` `itemsPerPage` prop

## 0.3.0
