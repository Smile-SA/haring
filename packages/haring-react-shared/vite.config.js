import { resolve } from 'path';

import react from '@vitejs/plugin-react-swc';
import { defineConfig } from 'vite';
import dts from 'vite-plugin-dts';

export default defineConfig({
  build: {
    lib: {
      entry: resolve(__dirname, 'src/index.tsx'),
      fileName: '[name]',
      formats: ['cjs', 'es'],
    },
    rollupOptions: {
      external: [
        '@mantine/carousel',
        '@mantine/core',
        '@mantine/dates',
        '@mantine/dropzone',
        '@mantine/hooks',
        '@phosphor-icons/react',
        '@smile/haring-react',
        '@smile/haring-react-dates',
        '@smile/haring-react-dropzone',
        '@smile/haring-react-shared',
        '@smile/haring-react-table',
        '@storybook/addon-actions',
        '@storybook/preview-api',
        '@tabler/icons-react',
        '@testing-library/react',
        'dayjs',
        'embla-carousel-react',
        'mantine-react-table',
        'react',
        'react/jsx-runtime',
      ],
      input: {
        index: resolve(__dirname, 'src/index.tsx'),
        server: resolve(__dirname, 'src/server.tsx'),
        'storybook-utils': resolve(__dirname, 'src/storybook-utils/index.ts'),
        'test-utils': resolve(__dirname, 'src/test-utils/index.ts'),
      },
      output: {
        banner: (chunkInfo) => {
          if (
            ['src/index.tsx'].find((modulePath) =>
              chunkInfo.facadeModuleId?.endsWith(modulePath),
            )
          ) {
            return `"use client"`;
          }
          return '';
        },
        preserveModules: false,
      },
    },
  },
  plugins: [
    react(),
    dts({
      insertTypesEntry: true,
      tsconfigPath: './tsconfig.build.json',
    }),
  ],
});
