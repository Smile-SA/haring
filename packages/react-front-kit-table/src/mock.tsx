/* eslint-disable react-refresh/only-export-components */
export * from './Components/Table/Table.mock';
export * from './Components/TableGridView/TableGridView.mock';
