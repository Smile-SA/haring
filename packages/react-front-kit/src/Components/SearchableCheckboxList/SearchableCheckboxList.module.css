.checkboxRoot {
  margin-bottom: 10px;
}

.checkboxBottom {
  border-top: 1px solid var(--mantine-color-gray-2);
  padding: 17px 0 24px;
}

.checkboxTop {
  margin: 16px 0;
}

/* TODO: Known bug with label not positioned to the right, and missing gap, fixed in next patch */
