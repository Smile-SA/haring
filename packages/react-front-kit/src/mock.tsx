/* eslint-disable react-refresh/only-export-components */

export * from './Components/ResponsiveTabs/ResponsiveTabs.mock';
export * from './Components/SidebarMenu/SidebarMenu.mock';
export * from './Components/SwitchableView/SwitchableView.mock';
export * from './Components/Thumbnail/Thumbnail.mock';
export * from './Components/ThumbnailGrid/ThumbnailGrid.mock';
export * from './Layouts/FoldableColumnLayout/FoldableColumnLayout.mock';
