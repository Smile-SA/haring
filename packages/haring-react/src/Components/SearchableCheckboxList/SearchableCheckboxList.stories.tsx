import type { Meta, StoryObj } from '@storybook/react';

import { SearchableCheckboxList as Cmp } from './SearchableCheckboxList';
import { checkboxesMock } from './SearchableCheckboxList.mock';

const meta = {
  component: Cmp,
  tags: ['autodocs'],
  title: '3-custom/Components/SearchableCheckboxList',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const SearchableCheckboxList: IStory = {
  args: {
    checkboxes: checkboxesMock,
  },
};
