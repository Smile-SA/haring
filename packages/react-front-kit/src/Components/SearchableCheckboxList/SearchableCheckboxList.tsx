'use client';

import type { ChangeEvent, ReactElement } from 'react';

import {
  Button,
  Checkbox,
  Divider,
  Input,
  useMantineTheme,
} from '@mantine/core';
import { MagnifyingGlass } from '@phosphor-icons/react';
import { useState } from 'react';

import { useStyles } from './SearchableCheckboxList.style';

export interface ICheckbox {
  active?: boolean;
  id: number | string;
  label: string;
}

export interface ISearchableCheckboxListProps<T extends ICheckbox> {
  buttonLabel?: string;
  checkboxes?: T[];
  onClickButton?: (checkboxes: T[]) => void;
  placeholder?: string;
}

export function SearchableCheckboxList<T extends ICheckbox>(
  props: ISearchableCheckboxListProps<T>,
): ReactElement {
  const {
    buttonLabel = 'Validate changes',
    placeholder = 'Search in options',
    checkboxes = [],
    onClickButton,
  } = props;
  const [searchInput, setSearchInput] = useState('');
  const [newCheckboxes, setNewCheckboxes] = useState(checkboxes);

  const theme = useMantineTheme();
  const { classes } = useStyles();

  function handleInputChange(event: ChangeEvent<HTMLInputElement>): void {
    setSearchInput(event.target.value);
  }

  function handleCheckboxChange(item: ICheckbox): void {
    setNewCheckboxes(
      newCheckboxes.map((element) => {
        if (element.id !== item.id) {
          return element;
        }
        element.active = !element.active;
        return element;
      }),
    );
  }

  function checkboxIncludesSearchInput(label: string): boolean {
    return label.toLocaleLowerCase().includes(searchInput.toLocaleLowerCase());
  }

  return (
    <>
      <Input
        onChange={handleInputChange}
        placeholder={placeholder}
        rightSection={
          <MagnifyingGlass
            color={theme.fn.primaryColor()}
            size={20}
            values={searchInput}
            weight="bold"
          />
        }
      />
      <div className={classes.checkboxsTop}>
        {checkboxes.map((item) => {
          if (!item.active && checkboxIncludesSearchInput(item.label)) {
            return (
              <Checkbox
                key={item.id}
                classNames={{ root: classes.checkboxRoot }}
                defaultChecked={item.active}
                label={item.label}
                onChange={() => handleCheckboxChange(item)}
              />
            );
          }
          return '';
        })}
      </div>
      <Divider color="gray.2" />
      <div className={classes.checkboxsBottom}>
        {checkboxes.map((item) => {
          if (item.active && checkboxIncludesSearchInput(item.label)) {
            return (
              <Checkbox
                key={item.id}
                classNames={{ root: classes.checkboxRoot }}
                defaultChecked={item.active}
                label={item.label}
                onChange={() => handleCheckboxChange(item)}
              />
            );
          }
          return '';
        })}
      </div>
      <Button
        fullWidth
        onClick={() => onClickButton?.(newCheckboxes)}
        size="md"
      >
        {buttonLabel}
      </Button>
    </>
  );
}
