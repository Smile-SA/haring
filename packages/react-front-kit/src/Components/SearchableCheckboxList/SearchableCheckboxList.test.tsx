import { renderWithProviders } from '@smile/haring-react-shared/test-utils';

import { SearchableCheckboxList } from './SearchableCheckboxList';
import { checkboxesMock } from './SearchableCheckboxList.mock';

describe('SearchableCheckboxList', () => {
  beforeEach(() => {
    // Prevent mantine random ID
    Math.random = () => 0.42;
  });
  it('matches snapshot', () => {
    const { container } = renderWithProviders(
      <SearchableCheckboxList
        buttonLabel="Test"
        checkboxes={checkboxesMock}
        placeholder="Test"
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
