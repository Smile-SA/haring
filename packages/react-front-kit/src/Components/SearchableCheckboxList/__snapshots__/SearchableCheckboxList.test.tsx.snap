// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchableCheckboxList matches snapshot 1`] = `
<div>
  <div
    class="mantine-Input-wrapper mantine-1v7s5f8"
  >
    <input
      aria-invalid="false"
      class="mantine-Input-input mantine-v2g2mm"
      placeholder="Test"
    />
    <div
      class="mantine-1nmvwx4 mantine-Input-rightSection"
    >
      <svg
        fill="#0b7285"
        height="20"
        values=""
        viewBox="0 0 256 256"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M232.49,215.51,185,168a92.12,92.12,0,1,0-17,17l47.53,47.54a12,12,0,0,0,17-17ZM44,112a68,68,0,1,1,68,68A68.07,68.07,0,0,1,44,112Z"
        />
      </svg>
    </div>
  </div>
  <div
    class="mantine-1o8kcxl"
  >
    <div
      class="mantine-Checkbox-root mantine-lqd391"
    >
      <div
        class="mantine-hqx0d1 mantine-Checkbox-body"
      >
        <div
          class="mantine-1vf457v mantine-Checkbox-inner"
        >
          <input
            class="mantine-1mibt2f mantine-Checkbox-input"
            id="mantine-f4bipx4bi"
            type="checkbox"
          />
          <svg
            class="___ref-icon mantine-q9qwbl mantine-Checkbox-icon"
            fill="none"
            viewBox="0 0 10 7"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M4 4.586L1.707 2.293A1 1 0 1 0 .293 3.707l3 3a.997.997 0 0 0 1.414 0l5-5A1 1 0 1 0 8.293.293L4 4.586z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div
          class="mantine-g88pth mantine-Checkbox-labelWrapper"
        >
          <label
            class="mantine-14fko7b mantine-Checkbox-label"
            for="mantine-f4bipx4bi"
          >
            Tomato
          </label>
        </div>
      </div>
    </div>
    <div
      class="mantine-Checkbox-root mantine-lqd391"
    >
      <div
        class="mantine-hqx0d1 mantine-Checkbox-body"
      >
        <div
          class="mantine-1vf457v mantine-Checkbox-inner"
        >
          <input
            class="mantine-1mibt2f mantine-Checkbox-input"
            id="mantine-f4bipx4bi"
            type="checkbox"
          />
          <svg
            class="___ref-icon mantine-q9qwbl mantine-Checkbox-icon"
            fill="none"
            viewBox="0 0 10 7"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M4 4.586L1.707 2.293A1 1 0 1 0 .293 3.707l3 3a.997.997 0 0 0 1.414 0l5-5A1 1 0 1 0 8.293.293L4 4.586z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div
          class="mantine-g88pth mantine-Checkbox-labelWrapper"
        >
          <label
            class="mantine-14fko7b mantine-Checkbox-label"
            for="mantine-f4bipx4bi"
          >
            Banana
          </label>
        </div>
      </div>
    </div>
  </div>
  <div
    class="mantine-Divider-root mantine-Divider-horizontal mantine-1rts92h"
    role="separator"
  />
  <div
    class="mantine-ggbtxo"
  >
    <div
      class="mantine-Checkbox-root mantine-lqd391"
    >
      <div
        class="mantine-hqx0d1 mantine-Checkbox-body"
      >
        <div
          class="mantine-1vf457v mantine-Checkbox-inner"
        >
          <input
            checked=""
            class="mantine-1mibt2f mantine-Checkbox-input"
            id="mantine-f4bipx4bi"
            type="checkbox"
          />
          <svg
            class="___ref-icon mantine-q9qwbl mantine-Checkbox-icon"
            fill="none"
            viewBox="0 0 10 7"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M4 4.586L1.707 2.293A1 1 0 1 0 .293 3.707l3 3a.997.997 0 0 0 1.414 0l5-5A1 1 0 1 0 8.293.293L4 4.586z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div
          class="mantine-g88pth mantine-Checkbox-labelWrapper"
        >
          <label
            class="mantine-14fko7b mantine-Checkbox-label"
            for="mantine-f4bipx4bi"
          >
            apple
          </label>
        </div>
      </div>
    </div>
    <div
      class="mantine-Checkbox-root mantine-lqd391"
    >
      <div
        class="mantine-hqx0d1 mantine-Checkbox-body"
      >
        <div
          class="mantine-1vf457v mantine-Checkbox-inner"
        >
          <input
            checked=""
            class="mantine-1mibt2f mantine-Checkbox-input"
            id="mantine-f4bipx4bi"
            type="checkbox"
          />
          <svg
            class="___ref-icon mantine-q9qwbl mantine-Checkbox-icon"
            fill="none"
            viewBox="0 0 10 7"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M4 4.586L1.707 2.293A1 1 0 1 0 .293 3.707l3 3a.997.997 0 0 0 1.414 0l5-5A1 1 0 1 0 8.293.293L4 4.586z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div
          class="mantine-g88pth mantine-Checkbox-labelWrapper"
        >
          <label
            class="mantine-14fko7b mantine-Checkbox-label"
            for="mantine-f4bipx4bi"
          >
            Orange 
          </label>
        </div>
      </div>
    </div>
  </div>
  <button
    class="mantine-UnstyledButton-root mantine-Button-root mantine-1t8lq1h"
    data-button="true"
    type="button"
  >
    <div
      class="mantine-1wpc1xj mantine-Button-inner"
    >
      <span
        class="mantine-1ryt1ht mantine-Button-label"
      >
        Test
      </span>
    </div>
  </button>
</div>
`;
